extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -888,7 +888,7 @@ private static String getVideoInfoUrl(final String id, final String sts) {
                         String streamUrl = tags.get("url");
                         // if video has a signature: decrypt it and add it to the url
                         if (tags.get("s") != null) {
-                            streamUrl = streamUrl + "&signature=" + decryptSignature(tags.get("s"), decryptionCode);
+                            streamUrl = streamUrl + "&" + tags.get("sp") + "=" + decryptSignature(tags.get("s"), decryptionCode);
```

Comment:
Thanks! I'd like to include this into 0.16.2 so we can get this fix out asap. Can we ensure that `tags.get("sp") != null`? If this is the case, we are good to go. Otherwise, I'd suggest to use `streamUrl = streamUrl + "&signature=" + decryptSignature(tags.get("s"), decryptionCode);` as fallback if `tags.get("sp") == null`.

Reply:
Ok, I'll implement and test it this evening